rmf_fleet_adapter/include/rmf_fleet_adapter/agv/FleetUpdateHandle.hpp

```diff
@@ -308,7 +313,8 @@ class FleetUpdateHandle : public std::enable_shared_from_this<FleetUpdateHandle>
     double recharge_threshold,
     double recharge_soc,
     bool account_for_battery_drain,
-    rmf_task::ConstRequestFactoryPtr finishing_request = nullptr);
+    rmf_task::ConstRequestFactoryPtr finishing_request = nullptr,
+    bool retreat_to_charger = true);
```

Comment:
I don't think this should be a bool. Instead it should be the timer period. If this is set to `std::nullopt`, we reset the timer thereby disabling the automatic retreat to charger behavior. This way we not only expose the ability to enable/disable this feature but also allow users to configure the rate at which the check happens. Else we will need to add another parameter for that.

The docstring above should also be updated and it's worth mentioning the timer there so users know how this check is performed.

```suggestion
    std::optional<rmf_traffic::Duration> check_retreat_to_charger_interval = rmf_traffic::time::from_seconds(10));
```